graylog2-web-interface/src/stores/connect.jsx

```diff
@@ -27,57 +38,77 @@ import { isFunction } from 'lodash';
  * connect(SamplesComponent, { samples: SamplesStore }, ({ samples }) => ({ samples: samples.filter(sample => sample.id === 4) }))
  *
  */
-export default (Component, stores, mapProps = props => props) => {
+
+function connect<Stores: Object, Props, ComponentType: React.ComponentType<Props>, MappedProps>(
+  Component: ComponentType,
+  stores: Stores,
+  mapProps: (ResultType<Stores>) => MappedProps = props => props,
+): React.ComponentType<$Diff<$Call<ExtractComponentProps, ComponentType>, MappedProps>> {
   const wrappedComponentName = Component.displayName || Component.name || 'Unknown/Anonymous';
-  class ConnectStoresWrapper extends React.Component {
-    constructor(props) {
-      super(props);
+  class ConnectStoresWrapper extends React.Component<$Diff<$Call<ExtractComponentProps, ComponentType>, MappedProps>> {
```

Comment:
I am currently seeing the following warning, when I open any search page:
```
ConnectStoresWrapper[Widget] stores=view: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop.
```

Reply:
✅